### ISSUE TITLE
Reading node types / schema creation fails when reading nodes with advanced data types

### ISSUE THREAD

**Issue by moxious** [neo4j-contrib/neo4j-spark-connector#182]
Use this test record in your Neo4j instance, this is intended to stress the typing system:

```
create (:Types { 
   strA: ["Hello"], 
   inteA: [4], 
   floA: [3.14], 
   boolA: [true], 
   ptA: [point({ latitude: 0, longitude: 0 })], 
   dt: date(), 
   tm: time(), 
   ldt: localdatetime(), 
   dtA: [date()], 
   tmA: [time()], 
   ldtA: [localdatetime()], 
   dt_t: datetime(), 
   lt:localtime(), 
   dur: duration.between(datetime(), date()) 
});
```

Now do this small pyspark program:

```
df = spark.read.format("org.neo4j.spark.DataSource") \
    .option("url", "bolt://neo4j:7687") \
    .option("authentication.basic.username", "neo4j") \
    .option("authentication.basic.password", "zeppelin") \
    .option("query", "MATCH (t:Types) RETURN t") \
    .load()

df.show()
```

Result:

```
Py4JJavaError: An error occurred while calling o284.showString.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 24.0 failed 1 times, most recent failure: Lost task 0.0 in stage 24.0 (TID 24, localhost, executor driver): java.lang.ClassCastException: org.apache.spark.unsafe.types.UTF8String cannot be cast to org.apache.spark.sql.catalyst.InternalRow
	at org.apache.spark.sql.catalyst.util.GenericArrayData.getStruct(GenericArrayData.scala:78)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.project_fieldToString_1_2$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$13$$anon$1.hasNext(WholeStageCodegenExec.scala:636)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$2.apply(SparkPlan.scala:255)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$2.apply(SparkPlan.scala:247)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
```

**Comment by utnaf** [comment]
Can't reproduce, tested with Neo4j 4.0.0, 4.0.8 and 4.1.1. Can you provide more info? Also the fact that APOCs are or aren't is important.

**Comment by moxious** [comment]
@utnaf I can reproduce at will.  It's neo4j 4.0.8 with APOC.   See branch "issue-reproduce" that I just pushed on the spark-connector-notebooks repo.   Open the "Play" workbook that I committed on that branch, and run the final paragraph.

If you can't reproduce, then it's down to the data in my local database and we can debug further.

**Comment by utnaf** [comment]
@moxious works for me

<img width="552" alt="Screenshot 2020-09-17 at 08 42 36" src="https://user-images.githubusercontent.com/1318539/93429662-cb375880-f8c1-11ea-87e4-2e7cb49862ce.png">

We can take a moment to debug later today

**Comment by utnaf** [comment]
Fixed by #195
